src/sparse/impl/KokkosSparse_spgemm_impl_triangle.hpp

```diff
@@ -1414,29 +1418,27 @@ void KokkosSPGEMM
 
   nnz_lno_t num_chunks = concurrency / suggested_vector_size;
   KokkosKernels::Impl::PoolType my_pool_type = KokkosKernels::Impl::OneThread2OneChunk;
-  if (MyEnumExecSpace == KokkosKernels::Impl::Exec_CUDA) {
+  if (exec_gpu) {
     my_pool_type = KokkosKernels::Impl::ManyThread2OneChunk;
   }
 
-
-#if defined( KOKKOS_ENABLE_CUDA )
-  size_t free_byte ;
-  size_t total_byte ;
-  cudaMemGetInfo( &free_byte, &total_byte ) ;
-  size_t required_size = size_t (num_chunks) * accumulator_chunksize * sizeof(nnz_lno_t);
-  if (KOKKOSKERNELS_VERBOSE)
-    std::cout << "\tmempool required size:" << required_size << " free_byte:" << free_byte << " total_byte:" << total_byte << std::endl;
-  if (required_size + num_chunks > free_byte){
-    num_chunks = ((((free_byte - num_chunks)* 0.5) /8 ) * 8) / sizeof(nnz_lno_t) / accumulator_chunksize;
-  }
-  {
-    nnz_lno_t min_chunk_size = 1;
-    while (min_chunk_size * 2 < num_chunks) {
-      min_chunk_size *= 2;
+  if(exec_gpu) {
```

Comment:
I saw this piece of code somewhere else, if I find it a third maybe we should factor it out as a utility?